Can't compile from master

I tried to compile from master from newest commit https://github.com/RobertBeckebans/RBDOOM-3-BFG/commit/f9ca463d508d03bd686b6fb56baa75b7ca28fe9e but I got error.
Arch Linux
From AUR https://aur.archlinux.org/packages/rbdoom3-bfg-git

Compile error:
https://gist.github.com/roland-5/b1976fef9ac6c8c6a1b860af96f8c320

I tried again, and had the same error:
```
[100%] Building CXX object CMakeFiles/RBDoom3BFG.dir/renderer/NVRHI/BufferObject_NVRHI.cpp.o
[100%] Building CXX object CMakeFiles/RBDoom3BFG.dir/renderer/NVRHI/Framebuffer_NVRHI.cpp.o
[100%] Building CXX object CMakeFiles/RBDoom3BFG.dir/renderer/NVRHI/Image_NVRHI.cpp.o
cc1plus: warning: /home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/idlib/precompiled.h.gch: created with 'none' debug info, but used with 'dwarf-2' [-Winvalid-pch]
cc1plus: warning: /home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/idlib/precompiled.h.gch: created with 'none' debug info, but used with 'dwarf-2' [-Winvalid-pch]
cc1plus: warning: /home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/idlib/precompiled.h.gch: created with 'none' debug info, but used with 'dwarf-2' [-Winvalid-pch]
[100%] Building CXX object CMakeFiles/RBDoom3BFG.dir/renderer/NVRHI/RenderBackend_NVRHI.cpp.o
cc1plus: warning: /home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/idlib/precompiled.h.gch: created with 'none' debug info, but used with 'dwarf-2' [-Winvalid-pch]
[100%] Building CXX object CMakeFiles/RBDoom3BFG.dir/renderer/NVRHI/RenderDebug_NVRHI.cpp.o
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp: In function ‘const char* Posix_Cwd()’:
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:753:15: warning: ignoring return value of ‘char* getcwd(char*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  753 |         getcwd( cwd, sizeof( cwd ) - 1 );
      |         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp: In function ‘void tty_Del()’:
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1145:14: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1145 |         write( STDOUT_FILENO, &key, 1 );
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1147:14: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1147 |         write( STDOUT_FILENO, &key, 1 );
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1149:14: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1149 |         write( STDOUT_FILENO, &key, 1 );
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp: In function ‘void tty_Left()’:
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1155:14: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1155 |         write( STDOUT_FILENO, &key, 1 );
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp: In function ‘void tty_Right()’:
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1161:14: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1161 |         write( STDOUT_FILENO, &key, 1 );
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1162:14: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1162 |         write( STDOUT_FILENO, "[C", 2 );
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp: In function ‘void tty_Show()’:
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1210:30: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1210 |                         write( STDOUT_FILENO, buf, strlen( buf ) );
      |                         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp: In function ‘char* Posix_ConsoleInput()’:
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/posix/posix_main.cpp:1275:46: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1275 |                                         write( STDOUT_FILENO, &key, 1 );
      |                                         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: warning: /home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/idlib/precompiled.h.gch: created with 'none' debug info, but used with 'dwarf-2' [-Winvalid-pch]
[100%] Building CXX object CMakeFiles/RBDoom3BFG.dir/renderer/NVRHI/RenderProgs_NVRHI.cpp.o
cc1plus: warning: /home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/idlib/precompiled.h.gch: created with 'none' debug info, but used with 'dwarf-2' [-Winvalid-pch]
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp: In member function ‘virtual bool DeviceManager_VK::CreateDeviceAndSwapChain()’:
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp:1274:26: error: ‘DynamicLoader’ in namespace ‘vk’ does not name a type; did you mean ‘DynamicState’?
 1274 |         static const vk::DynamicLoader dl;
      |                          ^~~~~~~~~~~~~
      |                          DynamicState
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp:1276:33: error: ‘dl’ was not declared in this scope
 1276 |         vkGetInstanceProcAddr = dl.getProcAddress<PFN_vkGetInstanceProcAddr>( "vkGetInstanceProcAddr" );
      |                                 ^~
/home/roland/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp:1276:76: error: expected primary-expression before ‘>’ token
 1276 |         vkGetInstanceProcAddr = dl.getProcAddress<PFN_vkGetInstanceProcAddr>( "vkGetInstanceProcAddr" );
      |                                                                            ^
make[2]: *** [CMakeFiles/RBDoom3BFG.dir/build.make:6575: CMakeFiles/RBDoom3BFG.dir/sys/DeviceManager_VK.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:283: CMakeFiles/RBDoom3BFG.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'rbdoom3-bfg-git-r2027.a51833e4-1':
checking dependencies...

Package (2)              Old Version  Net Change

directx-shader-compiler  1.8.2407-1   -38.58 MiB
rapidjson                1.1.0-6       -0.62 MiB
```

I got the same issue. I have even stripped out all C and CXX flags but the error is in the same place

[100%] Building CXX object CMakeFiles/RBDoom3BFG.dir/renderer/NVRHI/RenderProgs_NVRHI.cpp.o
/home/h3x/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp: In member function ‘virtual bool DeviceManager_VK::CreateDeviceAndSwapChain()’:
/home/h3x/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp:1274:26: error: ‘DynamicLoader’ in namespace ‘vk’ does not name a type; did you mean ‘DynamicState’?
 1274 |         static const vk::DynamicLoader dl;
      |                          ^~~~~~~~~~~~~
      |                          DynamicState
/home/h3x/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp:1276:33: error: ‘dl’ was not declared in this scope
 1276 |         vkGetInstanceProcAddr = dl.getProcAddress<PFN_vkGetInstanceProcAddr>( "vkGetInstanceProcAddr" );
      |                                 ^~
/home/h3x/.cache/paru/clone/rbdoom3-bfg-git/src/rbdoom3-bfg-git/neo/sys/DeviceManager_VK.cpp:1276:76: error: expected primary-expression before ‘>’ token
 1276 |         vkGetInstanceProcAddr = dl.getProcAddress<PFN_vkGetInstanceProcAddr>( "vkGetInstanceProcAddr" );
      |                                                                            ^
make[2]: *** [CMakeFiles/RBDoom3BFG.dir/build.make:6575: CMakeFiles/RBDoom3BFG.dir/sys/DeviceManager_VK.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:283: CMakeFiles/RBDoom3BFG.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...

--- a/neo/sys/DeviceManager_VK.cpp
+++ b/neo/sys/DeviceManager_VK.cpp
@@ -366,9 +366,9 @@ private:
 #endif
 
 private:
-	static VKAPI_ATTR VkBool32 VKAPI_CALL vulkanDebugCallback(
-		VkDebugReportFlagsEXT flags,
-		VkDebugReportObjectTypeEXT objType,
+	static VKAPI_ATTR vk::Bool32 VKAPI_CALL vulkanDebugCallback(
+		vk::DebugReportFlagsEXT flags,
+		vk::DebugReportObjectTypeEXT objType,
 		uint64_t obj,
 		size_t location,
 		int32_t code,
@@ -388,23 +388,23 @@ private:
 			}
 		}
 
-		if( flags & VK_DEBUG_REPORT_ERROR_BIT_EXT )
+		if( flags & vk::DebugReportFlagBitsEXT::eError )
 		{
 			idLib::Printf( "[Vulkan] ERROR location=0x%zx code=%d, layerPrefix='%s'] %s\n", location, code, layerPrefix, msg );
 		}
-		else if( flags & VK_DEBUG_REPORT_WARNING_BIT_EXT )
+		else if( flags & vk::DebugReportFlagBitsEXT::eWarning )
 		{
 			idLib::Printf( "[Vulkan] WARNING location=0x%zx code=%d, layerPrefix='%s'] %s\n", location, code, layerPrefix, msg );
 		}
-		else if( flags & VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT )
+		else if( flags & vk::DebugReportFlagBitsEXT::ePerformanceWarning )
 		{
 			idLib::Printf( "[Vulkan] PERFORMANCE WARNING location=0x%zx code=%d, layerPrefix='%s'] %s\n", location, code, layerPrefix, msg );
 		}
-		else if( flags & VK_DEBUG_REPORT_INFORMATION_BIT_EXT )
+		else if( flags & vk::DebugReportFlagBitsEXT::eInformation )
 		{
 			idLib::Printf( "[Vulkan] INFO location=0x%zx code=%d, layerPrefix='%s'] %s\n", location, code, layerPrefix, msg );
 		}
-		else if( flags & VK_DEBUG_REPORT_DEBUG_BIT_EXT )
+		else if( flags & vk::DebugReportFlagBitsEXT::eDebug )
 		{
 			idLib::Printf( "[Vulkan] DEBUG location=0x%zx code=%d, layerPrefix='%s'] %s\n", location, code, layerPrefix, msg );
 		}
@@ -640,7 +640,11 @@ void DeviceManager_VK::installDebugCallback()
 						   vk::DebugReportFlagBitsEXT::eWarning |
 						   //   vk::DebugReportFlagBitsEXT::eInformation |
 						   vk::DebugReportFlagBitsEXT::ePerformanceWarning )
+#if VK_HEADER_VERSION >= 304
 				.setPfnCallback( vulkanDebugCallback )
+#else
+				.setPfnCallback( reinterpret_cast<PFN_vkDebugReportCallbackEXT>( vulkanDebugCallback ) )
+#endif
 				.setPUserData( this );
 
 	const vk::Result res = m_VulkanInstance.createDebugReportCallbackEXT( &info, nullptr, &m_DebugReportCallback );
@@ -1251,13 +1255,20 @@ bool DeviceManager_VK::CreateDeviceAndSwapChain()
 	m_DeviceParams.enableNvrhiValidationLayer = r_useValidationLayers.GetInteger() > 0;
 	m_DeviceParams.enableDebugRuntime = r_useValidationLayers.GetInteger() > 1;
 
+	// SRS - DynamicLoader is in a separate namespace in newer versions of Vulkan-Hpp
+#if VK_HEADER_VERSION >= 301
+	using VulkanDynamicLoader = vk::detail::DynamicLoader;
+#else
+	using VulkanDynamicLoader = vk::DynamicLoader;
+#endif
+
 	if( m_DeviceParams.enableDebugRuntime )
 	{
 #if defined(__APPLE__) && defined( USE_MoltenVK )
 	}
 
 	// SRS - when USE_MoltenVK defined, load libMoltenVK vs. the default libvulkan
-	static const vk::DynamicLoader dl( "libMoltenVK.dylib" );
+	static const VulkanDynamicLoader dl( "libMoltenVK.dylib" );
 #else
 		enabledExtensions.layers.insert( "VK_LAYER_KHRONOS_validation" );
 
@@ -1276,7 +1287,7 @@ bool DeviceManager_VK::CreateDeviceAndSwapChain()
 	}
 
 	// SRS - make static so ~DynamicLoader() does not prematurely unload vulkan dynamic lib
-	static const vk::DynamicLoader dl;
+	static const VulkanDynamicLoader dl;
 #endif
 	vkGetInstanceProcAddr = dl.getProcAddress<PFN_vkGetInstanceProcAddr>( "vkGetInstanceProcAddr" );
 	VULKAN_HPP_DEFAULT_DISPATCHER.init( vkGetInstanceProcAddr );

